### PR TITLE
Check all CUDA API calls for errors in caffe2/

### PR DESCRIPTION
**Pull request by r-barnes**
Test Plan: Sandcastle

Differential Revision: D35194868

### CODE REVIEW

File: caffe2/core/context_gpu.h [hunk 1]
```diff
@@ -195,10 +195,6 @@ class CAFFE2_CUDA_API CUDAContext final : public BaseContext {
   // SwitchToDevice()
   void FinishDeviceComputation() override {
     CUDA_ENFORCE(cudaStreamSynchronize(getCudaObjects().GetStream(gpu_id_)));
-    cudaError_t error = cudaGetLastError();
-    if (error != cudaSuccess) {
-      CAFFE_THROW("Encountered CUDA error: ", cudaGetErrorString(error));
-    }
```
Comment (ezyang) [inline caffe2/core/context_gpu.h:201]: Hm?

**Review by ezyang — APPROVED** [review]

**Review by jeffdaily — COMMENTED** [review]
Why is CUDA_CHECK sometimes preferred over C10_CUDA_CHECK and vice versa?  Is there a clear reason to use one or the other?  Superficially, they seem inconsistently applied in this PR.

File: caffe2/core/context_gpu.h (hunk 1 above)
Comment (r-barnes) [reply caffe2/core/context_gpu.h:201]: All this is just a duplicate of what `CUDA_ENFORCE` is doing anyway.

**Comment by r-barnes** [comment]
@jeffdaily - I believe `CUDA_CHECK` is a deprecated way of performing the check while `C10_CUDA_CHECK` is more standard. I've switched to `C10_CUDA_CHECK` in most places now.

**Pull request closed without merging**